### ISSUE TITLE
GitHub REST API CORS error when using header X-GitHub-Api-Version with Octokit

### ISSUE THREAD

**Issue by johnnygerard** [github/docs#24706]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

[API Versions](https://docs.github.com/en/rest/overview/api-versions?apiVersion=2022-11-28)
[GitHub REST API documentation](https://docs.github.com/en/rest?apiVersion=2022-11-28)

### What changes are you suggesting?

The [GitHub REST API documentation](https://docs.github.com/en/rest?apiVersion=2022-11-28) indicates in various places the use of the X-GitHub-Api-Version header.

The X-GitHub-Api-Version header works with curl, GitHub CLI and Octokit on Node.js.
When using Octokit with a browser, I receive a CORS error (see below).

### Additional information

GitHub repository: [minimal reproducible example](https://github.com/johnnygerard/bug-github-version-api)
[Live test](https://bug-github-version-api.vercel.app/)

Error message:

`Access to fetch at 'https://api.github.com/users/octocat' from origin 'https://bug-github-version-api.vercel.app' has been blocked by CORS policy: Request header field x-github-api-version is not allowed by Access-Control-Allow-Headers in preflight response.`

**Comment by cmwilson21** [comment]
@johnnygerard Thanks so much for opening an issue! I'll triage this for the team to take a look :eyes:

**Comment by docubot** [comment]
Thank you for opening this issue! Updates to the REST API, GraphQL API, and webhook reference documentation must be made internally. I have copied your issue to an internal issue, so I will close this issue.

**Comment by skedwards88** [comment]
Apologies; I mistraiged this. I'm going to check in with the SDK team about this. You may also want to open an issue in the Octokit.js repo.

**Comment by kfcampbell** [comment]
Hi! My name is Keegan and I work on GitHub's SDK team. It's not super clear to me what's going on here. Do you mind elaborating a bit on "When using Octokit with a browser"? Are you still using Octokit.js or any associated plugins? My _guess_ would be that your Vercel site would need to change its CORS policy; Octokit.js doesn't deal with CORS directly itself.

**Comment by johnnygerard** [comment]
Hello @kfcampbell.

I tested the @octokit/request npm package both with Node.js and in the browser with Angular.

With Angular, it does not work. I also tried to use the Angular HttpClient instead of Octokit and I get the same CORS error message.

**Comment by johnnygerard** [comment]
Paste and run the following code in your browser developer tools console.
Toggle the comment to test with and without the version header.

```javascript
const URL = 'https://api.github.com/users/octocat';
const headers = {
  Accept: 'application/vnd.github+json',
  // 'X-GitHub-Api-Version': '2022-11-28',
};
const response = await fetch(URL, { headers });
const data = await response.json();
console.log(data);
```

**Comment by Kaminem2021** [comment]
‫‪Johnny Gérard‬‏ ***@***.***‬‏> در تاریخ شنبه ۱ آوریل ۲۰۲۳
ساعت ۵:۴۳ نوشت:‬

**Comment by gr2m** [comment]
This is an API problem. GitHub's API is CORS-enabled, however the new `X-GitHub-Api-Version` needs to be added to the explicitly permitted headers to be sent cross origins, using the [`Access-Control-Expose-Headers`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers) header

GitHub REST API's current value for the `Access-Control-Expose-Headers` header is

```
ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset
```

We will need to add the `X-GitHub-Api-Version` to the list.

**Comment by skedwards88** [comment]
This should now be fixed, so I'm going to close out this issue. Thanks everyone for reporting/responding!

**Comment by nguyenalex836** [comment]
Howdy, @Crystal-Spider! 👋 This question might be best handled by our wonderful [support team](https://support.github.com/request). We recommend reaching out to them, and you are welcome to reference this closed issue when you do!

If there's anything in the documents that you have suggestions for updating, please feel free to open a new issue in this repo 💛
